spec/features/idv/end_to_end_idv_spec.rb

```diff
@@ -276,10 +276,32 @@ def validate_review_submit(user)
   def validate_come_back_later_page
     expect(page).to have_current_path(idv_letter_enqueued_path)
     expect_in_person_gpo_step_indicator_current_step(t('step_indicator.flows.idv.get_a_letter'))
+    expect(page).to have_content(t('idv.titles.come_back_later'))
+    expect(page).not_to have_content(t('step_indicator.flows.idv.verify_phone_or_address'))
   end
 
   def validate_personal_key_page
     expect(current_path).to eq idv_personal_key_path
+
+    # Clicking acknowledge checkbox is required to continue
+    click_continue
+    expect(page).to have_content(t('forms.validation.required_checkbox'))
+    expect(current_path).to eq(idv_personal_key_path)
+
+    expect(page).to have_content(t('forms.personal_key_partial.acknowledgement.header'))
+    expect(page).to have_content(t('forms.personal_key_partial.acknowledgement.text'))
+    expect(page).to have_content(t('forms.personal_key_partial.acknowledgement.help_link_text'))
+    expect(page).to have_content(t('idv.messages.confirm'))
+    expect_step_indicator_current_step(t('step_indicator.flows.idv.secure_account'))
+    expect(page).to have_css(
+      '.step-indicator__step--complete',
+      text: t('step_indicator.flows.idv.verify_phone_or_address'),
+    )
+    expect(page).not_to have_content(t('step_indicator.flows.idv.get_a_letter'))
+
+    # Refreshing shows same page (BUT with new personal key, we should warn the user)
```

Comment:
Is this suggesting that the personal key is being regenerated each time? That would be surprising to me.

Is it not pulling from the `idv_session` here then?

https://github.com/18F/identity-idp/blob/ee5275ccc9a54973bee4a05a7a624695ecd545bb/app/controllers/idv/personal_key_controller.rb#L72-L74

Reply:
Huh, I didn't realize `idv_session` had a `personal_key` property. It's never set (except in specs). That would be an easy fix, if it's legit for security reasons to save a personal key in session.

Reply:
We have [LG-11221](https://cm-jira.usa.gov/browse/LG-11221) to fix this, and I just added a comment with what you pointed out, thanks.